master/setup.py

```diff
@@ -415,113 +416,98 @@ def define_plugin_entries(groups):
         if LooseVersion(pip_dist.version) < LooseVersion('1.4'):
             raise RuntimeError(VERSION_MSG)
 
-if sys.version_info[:2] == (2, 6):
-    # Twisted-15.4.0, txaio >=2.3.0 and autobahn >=0.13.0 don't support Python
-    # 2.6 anymore
-    twisted_ver = ">= 14.0.1, < 15.4.0"
-    autobahn_ver = ">= 0.10.2, < 0.13.0"
-    txaio_ver = "== 2.2.2"
-
-else:
-    twisted_ver = ">= 14.0.1"
-    autobahn_ver = ">= 0.16.0"
-    txaio_ver = ">= 2.2.2"
+twisted_ver = ">= 14.0.1"
+autobahn_ver = ">= 0.16.0"
+txaio_ver = ">= 2.2.2"
 
 bundle_version = version.split("-")[0]
 
-try:
-    # If setuptools is installed, then we'll add setuptools-specific arguments
-    # to the setup args.
-    import setuptools
-    [setuptools]
-except ImportError:
-    pass
-else:
-    # dependencies
-    setup_args['install_requires'] = [
-        'Twisted ' + twisted_ver,
-        'Jinja2 >= 2.1',
-        # required for tests, but Twisted requires this anyway
-        'zope.interface >= 4.1.1',
-        # python-future required for py2/3 compatibility
-        'future',
-        'sqlalchemy>=0.8.0',
-        'sqlalchemy-migrate>=0.9',
-        'python-dateutil>=1.5',
-        'txaio ' + txaio_ver,
-        'autobahn ' + autobahn_ver,
+# dependencies
+setup_args['install_requires'] = [
+    'setuptools >= 8.0',
+    'Twisted ' + twisted_ver,
+    'Jinja2 >= 2.1',
+    # required for tests, but Twisted requires this anyway
+    'zope.interface >= 4.1.1',
+    # python-future required for py2/3 compatibility
+    'future',
+    'sqlalchemy>=0.8.0',
+    'sqlalchemy-migrate>=0.9',
+    'python-dateutil>=1.5',
+    'txaio ' + txaio_ver,
+    'autobahn ' + autobahn_ver,
+]
+
+# Unit test dependencies.
+test_deps = [
+    # http client libraries
+    'treq',
+    'txrequests',
+    # pyjade required for custom templates tests
+    'pyjade',
+    # boto3 and moto required for running EC2 tests
+    'boto3',
+    'moto',
+    # txgithub required to run buildbot.status.github module tests
+    'txgithub',
+    'ramlfications',
+    'mock>=2.0.0',
+]
+if sys.platform != 'win32':
+    test_deps += [
+        # LZ4 fails to build on Windows:
+        # https://github.com/steeve/python-lz4/issues/27
+        # lz4 required for log compression tests.
+        'lz4',
     ]
 
-    # Unit test dependencies.
-    test_deps = [
-        # http client libraries
-        'treq',
-        'txrequests',
-        # pyjade required for custom templates tests
-        'pyjade',
-        # boto3 and moto required for running EC2 tests
-        'boto3',
-        'moto',
-        # txgithub required to run buildbot.status.github module tests
-        'txgithub',
+setup_args['tests_require'] = test_deps
+
+setup_args['extras_require'] = {
+    'test': [
+        'setuptools_trial',
```

Comment:
Actually `setup.py test` works, but it can not handle side-by-side installation of `autobahn` due to `setuptools` bug: https://github.com/pypa/setuptools/issues/500
So if current Python environment have `autobahn` installed, `setup.py test` should handle side-by-side installation of other deps and properly run tests.